src/cryptoadvance/specter/templates/includes/helpers.jinja

```diff
@@ -1,26 +1,198 @@
+        // use thousandsSeparator and decimalSeparator to simplify future generalization of locale
+        let thousandsSeparator = Number(1000).toLocaleString(locale).charAt(1);
+        let decimalSeparator = Number(1.1).toLocaleString(locale).charAt(1);
+
+        var countDigitsThatCanBeStripped = 0;
+        for (var j in array){
+            var i = array.length - j - 1;
+            if (array[i] == "0"){
+                countDigitsThatCanBeStripped += 1;
+                continue
+            }
+            break
+        }
+
+        if (countDigitsThatCanBeStripped >= minimumDigitsToStrip){
+            // loop through the float number, e.g. 0.03 000 000, from the right and replace 0's or the '.' until you hit anything != 0
+            for (var j in array){
+                var i = array.length - j - 1;
+                if ((array[i] == "0") && (array.length - i <= maximumDigitsToStrip)){
+                    array[i] = `<span class="unselectable transparent-text ${hideStripped ? 'hidden' : ''}">${array[i]}</span>`;
+                    // since this digit == 0, then continue the loop and check the next digit
+                    continue
+                }
+                // the following if branch is only relevant if last_digits_to_strip == 8, i.e. all digits can be stripped
+                else if (formatted_amount[i] == decimalSeparator){
+                    array[i] = `<span class="unselectable transparent-text ${hideStripped ? 'hidden' : ''}">${array[i]}</span>`;
+                    // since this character == '.', then the loop must be broken now
+                }
+                // always break the loop. Only the digit == 0 can prevent this break
+                break
+            }
+        }
+
+        const l = array.length;
+        if (enableDigitFormatting){
+            array[l-6] = `<span class="thousand-digits-in-btc-amount">${array[l-6]}`;
+            array[l-4] = `${array[l-4]}</span>`;
+            array[l-3] = `<span class="last-digits-in-btc-amount">${array[l-3]}`;
+            array[l-1] = `${array[l-1]}</span>`;
+        }
+
+        return array.join('');
+    }
+
+
+    // Puts the price in a "note" span.  preText is separator text 
+    function internalFormatPriceAsNote(price, preText=''){
+        return `<span class="note">${preText}(${price})</span>`
+    }
+
+
+    // Determines if an unit (e.g. from a tx output) is "BTC", "LBTC", "tBTC", "tLBTC"
+    // null or "" will also return true
+    function internalUnitLabelIsBitcoin(unit){
+        if (!unit){
+            return true            
+        } 
+        unit = unit.toUpperCase();
+        return ([null, "", "LBTC", "BTC", "TBTC", "TLBTC"].indexOf(unit) > -1)
+    }
+
+
+
+
+    // Formats the network independent units to network dependent units,
+    // e.g.: btc -> tBTC,  sat --> tsat ,  btc --> tLBTC, sat --> tLsat 
+    function formatUnitLabel(targetUnit='{{ "BTC" if specter.is_liquid else specter.unit }}'){ 
```

Comment:
`targetUnit={{ "BTC" if specter.is_liquid else specter.unit }}` is defined multiple times here, would improve readability IMO to only set `targetUnit` in the outer functions: `formatBtcAmountAndUnit` and `formatBtcAmount`. Also what is the point of` "BTC" if specter.is_liquid`? Just check the `specter.unit` with Liquid is the same as with First Layer Bitcoin.

Reply:
Thanks for addressing this.  My current approach is:

- If it is not liquid, then this function will convert the number to 'sat' or keep 'BTC' depending on `specter.unit`
- if it is liquid, then targetUnit="BTC"  will prevent any conversion to sat, regardless of `specter.unit`. I currently do not know how to generate such a unit label ``<asset-label data-asset="${Object.keys(this.addressData.assets)[0]}"></asset-label>`` in `formatUnitLabel`.

But I agree the current structure is not self-explanatory. I will put some thought in to handle the liquid case better.

Reply:
> Perhaps it would be best to separate Liquid in completely own functions.

Yeah...... I think that would be cleanest...     I think it is best you pause the review until I separated liquid out.

Reply:
Separation is always good. I will work in parallel to figure out how we best test all this. I was thinking to use Cypress to do some JS unit tests there similar to what you did in the Notifications PR:
<img width="679" alt="grafik" src="https://user-images.githubusercontent.com/70536101/192587140-35816cd3-95ac-4dcf-985f-364da390f947.png">

Reply:
> I will work in parallel to figure out how we best test all this.

thanks!  I have extended the cypress test, but a cypress unit test would be better.

I separated the Liquid part in own functions. I think everything works and is ready for review.